tests/Utils/FileSystem.phpt

```diff
@@ -41,8 +41,13 @@ test(function () { // copy
 	FileSystem::copy(TEMP_DIR . '/3/file', TEMP_DIR . '/3/x/file');
 	Assert::same('Hello', file_get_contents(TEMP_DIR . '/3/x/file'));
 
-	FileSystem::copy('http://example.com', TEMP_DIR . '/3/x/y/file');
-	Assert::true(is_file(TEMP_DIR . '/3/x/y/file'));
+	if (getenv('SKIP_ONLINE_TESTS') ?: false) {
+		FileSystem::copy(__FILE__, TEMP_DIR . '/3/x/y/file');
+		Assert::true(is_file(TEMP_DIR . '/3/x/y/file'));
+	} else {
+		FileSystem::copy('http://example.com', TEMP_DIR . '/3/x/y/file');
+		Assert::true(is_file(TEMP_DIR . '/3/x/y/file'));
+	}
```

Comment:
Why not just

~~~php
if (!getenv('SKIP_ONLINE_TESTS')) {
    FileSystem::copy('http://example.com', TEMP_DIR . '/3/x/y/file');
    Assert::true(is_file(TEMP_DIR . '/3/x/y/file'));
}
~~~
?